src/dbcp/transform/gridstatus.py

```diff
@@ -0,0 +1,632 @@
+    )
+    return iso_df
+
+
+def transform(raw_dfs: dict[str, pd.DataFrame]) -> dict[str, pd.DataFrame]:
+    """Clean Grid Status Interconnection Queue data.
+
+    Args:
+        raw_dfs: raw dataframes for each ISO.
+
+    Returns:
+        A dictionary of cleaned Grid Status data queus.
+    """
+    # create one dataframe
+    shared_columns_mapping = {
+        "Actual Completion Date": "actual_completion_date",
+        "Capacity (MW)": "capacity_mw",
+        "County": "county",
+        "Generation Type": "resource",
+        "Interconnecting Entity": "interconnecting_entity",
+        "Interconnection Location": "point_of_interconnection",
+        "Project Name": "project_name",
+        "Proposed Completion Date": "proposed_completion_date",
+        "Queue Date": "queue_date",
+        "Queue ID": "queue_id",
+        "State": "state",
+        "Status": "queue_status",
+        "Summer Capacity (MW)": "summer_capacity_mw",
+        "Transmission Owner": "transmission_owner",
+        "Winter Capacity (MW)": "winter_capacity_mw",
+        "Withdrawal Comment": "withdrawal_comment",
+        "Withdrawn Date": "withdrawn_date",
+        "is_actionable": "is_actionable",
+        "is_nearly_certain": "is_nearly_certain",
+        "iso_region": "iso_region",
+    }
+
+    iso_cleaning_functions = {
+        "miso": _transform_miso,
+        "caiso": _transform_caiso,
+        "pjm": _transform_pjm,
+        "ercot": _transform_ercot,
+        "spp": _transform_spp,
+        "nyiso": _transform_nyiso,
+        "isone": _transform_isone,
+    }
+
+    projects = []
+    for iso, df in raw_dfs.items():
+        # Apply rename
+        renamed_df = df.rename(columns=shared_columns_mapping).copy()
+
+        # Apply iso specific cleaning functions
+        renamed_df = iso_cleaning_functions[iso](renamed_df)
+
+        renamed_df["iso_region"] = iso
+        renamed_df = renamed_df[shared_columns_mapping.values()]
+        projects.append(renamed_df)
+
+    projects = pd.concat(projects)
+    projects["queue_status"] = projects.queue_status.str.lower()
+    active_projects = projects.query("queue_status == 'active'").copy()
+
+    # parse dates
+    date_cols = [col for col in list(projects) if "date" in col]
+    for col in date_cols:
+        active_projects[col] = pd.to_datetime(active_projects[col])
+
+    # create project_id
+    active_projects["project_id"] = np.arange(len(active_projects), dtype=np.int32)
+    active_projects = active_projects.set_index("project_id")
+
+    # geocode the projects
+    active_projects = add_county_fips_with_backup_geocoding(
+        active_projects, state_col="state", locality_col="county"
+    )
+    # correct some fips codes
+    active_projects.loc[
+        active_projects.county_id_fips.eq("51515"), "county_id_fips"
+    ] = "51019"  # https://www.ddorn.net/data/FIPS_County_Code_Changes.pdf
+
+    # harmonize types
+    active_projects = _clean_resource_type(active_projects)
```

Comment:
in the LBNL data, hybrid projects were modeled as 1:m relationships between projects:resources. This is most obvious in CAISO, where there are columns like resource_1, resource_2 and capacity_1, capacity_2. But other ISOs that label them like "solar; storage" were modeled the same way. If they don't give separate capacity values, it was just duplicated.

Reply:
Yeah I didn't get around to sorting out the 1:m relationship in CAISO. I think this iteration of integration just grabs the `*_1` values for CAISO projects.